src/core/src/package_managers/ZypperPackageManager.py

```diff
@@ -79,6 +85,68 @@ def invoke_package_manager(self, command):
             self.set_package_manager_setting(Constants.PACKAGE_MGR_SETTING_REPEAT_PATCH_OPERATION, True)
         return out
 
+    def get_process_tree_from_pid_in_output(self, message):
+        """ When given a string containing a pid, returns a multi-line string with a list of parent/child processes for that pid.
+            See pattern matching inside the function for details on specific strings that are accepted.
+
+            Example:
+                input:
+                    message (string): Output from package manager: | System management is locked by the application with pid 7914 (/usr/bin/zypper).
+
+                returns (string):
+                      PID CMD
+                     7736 /bin/bash
+                     7912  \_ python3 package_test.py
+                     7913  |   \_ sudo LANG=en_US.UTF8 zypper --non-interactive update --dry-run bind-utils
+                     7914  |       \_ zypper --non-interactive update --dry-run bind-utils
+                     7982  |           \_ /usr/bin/python3 /usr/lib/zypp/plugins/urlresolver/susecloud
+                     7984  |               \_ /usr/bin/python3 /usr/bin/azuremetadata --api latest --subscriptionId --billingTag --attestedData --signature
+                     7986  \_ python3 package_test.py
+                     8298      \_ sudo LANG=en_US.UTF8 zypper --non-interactive update --dry-run grub2-i386-pc
+        """
+
+        """ First find pid xxxxx within output string.
+
+            Example: 'Output from package manager: | System management is locked by the application with pid 7914 (/usr/bin/zypper).'
+
+            pid_substr_search will contain: ' pid 7914'
+        """
+        regex = re.compile(' pid \d+')
+        pid_substr_search = regex.search(message)
+        if pid_substr_search is None:
+            return None
+
+        """ Now extract just pid text from pid_substr_search.
+            
+            Example (pid_substr_search): ' pid 7914'
+                
+            pid_search will contain: '7914'
+        """
+        regex = re.compile('\d+')
+        pid_search = regex.search(pid_substr_search.group())
+        if pid_search is None:
+            return None
+
+        pid = pid_search.group()
+
+        # TODO: Check to make sure the pid is actually a running process
+        # this is because the below command will always return something, even for nonexistent pids
+
+        # Gives a process tree so the calling process name(s) can be identified
+        # TODO: consider revisiting in the future to reduce the result to this pid only instead of entire tree
+        get_process_tree_cmd = self.zypper_get_process_tree_cmd.format(str(pid))
+        code, out = self.env_layer.run_command_output(get_process_tree_cmd, False, False)
+
+        # Failed to get process tree
+        if code != 0:
+            return None
+
+        # Also failed to get process tree
+        if len(out) == 0:
```

Comment:
the 2 ifs can be combined into one:

if code !=0 or len(out) ==0:
   return None